widgets: Make the todo and poll widgets look consistent / similar.

N-Shar-ma opened the pull request:
Earlier the todo and poll widgets' UI differed a lot, especially in the
font size and weight, and the color and shape of the checkbox and vote
count box.

Now the font weights and sizes are consistent in both the widgets and
the todo widget's checkboxes and poll widget's vote count boxes are
styled similarly.

This is the rebased and cleaned version of #21006, and is visually the
same. (The commit history has been tidied and unnecessary naming
changes have been discarded, so we only have the minimal necessary
CSS only changes.)

PR #22551 also makes some related UI (though mostly functional) changes.

Fixes: #20283.

**Screenshots and screen captures:**

![image](https://user-images.githubusercontent.com/68962290/188196478-bb01cb46-6bb7-44f7-8411-f03b022d28c8.png)

![image](https://user-images.githubusercontent.com/68962290/188196544-65a4a33e-f9e3-4eac-b400-cf212a45291c.png)

GIFs can be seen in the original PR (#21006)

**Self-review checklist**

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.

alya commented:
@YashRE42 , since you reviewed #21006 earlier, would you be up for doing a round of review here?

alya commented:
@YashRE42 any feedback on this one? Thanks!

N-Shar-ma commented:
Just commenting to bump this up since @YashRE42 seems to have been inactive for a while now

amanagr commented:
@N-Shar-ma thanks for working on this! Left some comments!

N-Shar-ma commented on static/styles/widgets.css:
```diff
@@ -20,14 +20,75 @@
-        margin-left: 2px;
+    .add-task,
+    .add-desc {
+        font-size: 10pt;
```
The reason for using `pt` is that the corresponding code for the poll widget also used `pt` for these particular property values, and to keep the looks of the widgets (and their code too) consistent, I (or rather the original PR) used the same values and units.

Do you still think we should change them to `px`? If so, I think there can be a separate commit, or maybe another PR, porting both poll and todo widgets to using just `px` and not `pt`

N-Shar-ma replied:
Some of the font sizes in the current main too are in `pt` though... Do I change those too?

N-Shar-ma replied:
@amanagr Have replaced all instances of `pt` in widgets.css with their closest `px` equivalents. The UI is almost pixel wise identical as before... This should pass review now!

N-Shar-ma commented on static/styles/widgets.css:
```diff
@@ -55,25 +116,55 @@
         margin-right: 4px;
         padding-left: 2px; /* padding for Chromium browsers */
         padding-right: 2px;
-        min-width: 28px;
+        min-width: 22px;
+        min-height: 22px;
```
These again are from the original PR and they just happen to look the best I think (like to keep it square)

N-Shar-ma commented:
Noticed that visual indicators for hover state were missing in the todo widget for dark mode, so added those:

https://user-images.githubusercontent.com/68962290/205689850-bcc0cf0d-2032-4ce3-b473-681a2742dc65.mp4

alya commented:
@amanagr Any more feedback from your side? I'll re-review once the PR is ready from your perspective. Thanks!

amanagr commented:
@N-Shar-ma thanks for the update! Can you split the changes to poll widget and todo widget into independent commits. Also, it would be helpful if you can split css changes that are independent of other changes like conversion of `pt` to `px` values for the poll widget.

N-Shar-ma commented:
@amanagr I've separated a commit for conversion of `pt` to `px`. Since the issue involves making both the widgets looks similar, and many of the CSS changes apply to both widgets, I'm not sure if we should separate these changes into 2 commits when everything is so closely related... I hope that makes sense?

amanagr commented:
@N-Shar-ma LGTM! Would it make sense to make the size of checkbox in todo and numbers in poll widget the same?

alya commented:
> Would it make sense to make the size of checkbox in todo and numbers in poll widget the same?

I think we should keep the checkboxes square, but it would be very reasonable to try making the poll boxes square rather than rectangular too. I'm not sure how to easily test polls with >9 votes for an option, but we should make sure multi-digit vote totals continue to work fine.

N-Shar-ma commented:
@alya, I tweaked the height and width for the checkboxes just a lil so that single and double digit votes fit comfortably in a square shape, while checkboxes with >=3 digits votes retain their initial height but widen to the right, to allow the vote count to fit horizontally:
![image](https://user-images.githubusercontent.com/68962290/210975314-06fd491d-7317-47f0-9675-0e9bc78fe2f0.png)

Does this look okay?

alya commented:
Looks great to me!

alya commented:
@timabbott This is a major refresh on the appearance of polls and todo lists that I'd love to get integrated. It's been reviewed by @amanagr and me.

timabbott commented on static/styles/dark_theme.css:
```diff
@@ -328,7 +328,15 @@
+                .poll-vote,
+                button.poll-option,
+                button.add-task
+            ) {
```
This `not` block is really messy, and we should figure out a way to refactor it (probably involving these buttons not having the `new-style .button` class applied to them), but that's an existing mess and probably best left for future work.

timabbott commented on static/styles/zulip.css:
```diff
@@ -580,7 +580,7 @@ label {
 /* Bootstrap's focus outline uses -webkit-focus-ring-color which doesn't exist in Firefox */
 a:not(:active):focus,
-button:focus,
+button:not(.poll-option, .add-task, .poll-vote):focus,
```
Can we avoid this by just setting the `outline` property specifically for the new buttons? I don't want to add a `not` here.

N-Shar-ma replied:
Also removed another similar `not` block in `zulip.css` which defined outline color

timabbott commented:
I posted a couple comments; otherwise the code looks reasonable to me. I'll do manual testing and hopefully integrate once the above are resolved. Thanks @N-Shar-ma!

N-Shar-ma commented:
@timabbott Have addressed your comments, and also removed duplication in the styling of `.add-task` and `.poll-option` buttons across the 2 widgets, which I'd somehow missed till now.

timabbott commented:
Merged, thanks @N-Shar-ma! I think we're going to want to do a proper redesign pass on these widgets, but that doesn't mean this isn't a significant visual improvement here :).
